scripts/generate_openapi_document.py

```diff
@@ -0,0 +1,35 @@
+import json
+import os
+from unittest import (
+    mock,
+)
+
+from azul import (
+    config,
+)
+from azul.files import (
+    write_file_atomically,
+)
+from azul.modules import (
+    load_app_module,
+)
+
+
+def main():
+    mock_catalogs = mock.patch.dict(os.environ,
+                                    AZUL_CATALOGS='hca:dcp2:repository/tdr:metadata/hca')
+    mock_service_endpoint = mock.patch.object(target=config,
+                                              attribute='service_endpoint',
+                                              return_value='localhost')
+    mock_config_service_name = mock.patch.object(target=config,
+                                                 attribute='service_function_name',
+                                                 return_value='azul_service')
+    with mock_catalogs, mock_config_service_name, mock_service_endpoint:
```

Comment:
```diff
Index: scripts/generate_openapi_document.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/scripts/generate_openapi_document.py b/scripts/generate_openapi_document.py
--- a/scripts/generate_openapi_document.py	(revision 445cd50de9a6b9b3893b0f77583ab7fc7eeae30a)
+++ b/scripts/generate_openapi_document.py	(date 1624312407376)
@@ -1,7 +1,7 @@
 import json
 import os
-from unittest import (
-    mock,
+from unittest.mock import (
+    patch,
 )
 
 from azul import (
@@ -16,15 +16,16 @@
 
 
 def main():
-    mock_catalogs = mock.patch.dict(os.environ,
-                                    AZUL_CATALOGS='hca:dcp2:repository/tdr:metadata/hca')
-    mock_service_endpoint = mock.patch.object(target=config,
-                                              attribute='service_endpoint',
-                                              return_value='localhost')
-    mock_config_service_name = mock.patch.object(target=config,
-                                                 attribute='service_function_name',
-                                                 return_value='azul_service')
-    with mock_catalogs, mock_config_service_name, mock_service_endpoint:
+    with (
+        patch.dict(os.environ,
+                   AZUL_CATALOGS='hca:dcp2:repository/tdr:metadata/hca'),
+        patch.object(target=config,
+                     attribute='service_function_name',
+                     return_value='azul_service'),
+        patch.object(target=config,
+                     attribute='service_endpoint',
+                     return_value='localhost')
+    ):
         app_module = load_app_module('service')
         doc_path = os.path.join(config.project_root, 'lambdas/service/openapi.json')
         with write_file_atomically(doc_path) as file:
```

Reply:
mhm. cant use a tuple here because of an attribute error....
```
python /Users/amar/dev/hca/azul/scripts/generate_openapi_document.py
Traceback (most recent call last):
  File "/Users/amar/dev/hca/azul/scripts/generate_openapi_document.py", line 36, in <module>
    main()
  File "/Users/amar/dev/hca/azul/scripts/generate_openapi_document.py", line 19, in main
    with (
AttributeError: __enter__
make: *** [openapi] Error 1
```

Reply:
@hannes-ucsc tuples within `with` statements wont work until 3.10 lands. https://www.python.org/dev/peps/pep-0617/#migration-plan

Reply:
That link is a bit unspecific. I'm assuming you're referring to this bug: 

https://bugs.python.org/issue12782

Reply:
Just nest with statements in that case. The indirection through long variable names is unwieldy.